### ISSUE TITLE
Best practices for spectrum data loading

### ISSUE THREAD

**Issue by bittremieux** [wfondrie/depthcharge#62]
What's the recommend way to load spectra with a `SpectrumDataset`? Specifically:

1. I want to know the total size of the dataset to enable progress estimates (using `log(..., prog_bar=True)` from PyTorch Lightning). #60 
2. I want to use `shuffle=True` in the training `DataLoader` to avoid that batches are seen in the same order across different epochs. However, shuffling seems incompatible with an `IterableDataloader`.
3. I want to efficiently load batches of multiple spectra. The documentation mentions the following:
> The `batch_size` parameter for this class indepedent of the `batch_size` of the PyTorch DataLoader. Generally, we only want the former parameter to greater than 1. Additionally, this dataset should not be used with a DataLoader set to `max_workers` > 1, unless specific care is used to handle the [caveats of a PyTorch IterableDataset](https://pytorch.org/docs/stable/data.html#torch.utils.data.IterableDataset)

What's then the recommended number of workers in the `SpectrumDataset` and the `DataLoader`? The number of spectra in our batch in the former and `None` in the latter?
4. However, this setup is not compatible with doing `drop_last=True` in the `DataLoader` to avoid that incomplete batches are used, for which `batch_size` in the `DataLoader` needs to be set.
